### PR TITLE
Spec: Add fetch()-style API as a considered alternative

### LINKED ISSUE #42: Fetch based design as a considered alternative
When working on [Pending Beacon](https://github.com/WICG/pending-beacon), we got [feedback](https://github.com/WICG/pending-beacon/issues/70) that the design should be as close to `fetch()` as possible. There are parallels here, so I'm filing this issue as means to think through the implications of that alternative design.

A few differences between fetch()/PendingBeacon and this API:
* It's an anti-goal here to give the developer any control over when reports are being sent or knowledge that they were sent
* The reports are sent to very specific end-points, and not to any arbitrary reporting end-point
* The report's input is extremely specific (an array of [PAHistogramContribution](https://patcg-individual-drafts.github.io/private-aggregation-api/#dictdef-pahistogramcontribution)s), and not amenable to `fetch()`'s general purpose sending
* There's no concept of a response (even though that's similar for PendingBeacon)

Given all the above, I think it makes sense to not try to align this with `fetch()`. At the same time, it's worthwhile to document the reasoning.

### PR DESCRIPTION
**Pull request by alexmturner**
Resolves #42.

***
<a href="https://pr-preview.s3.amazonaws.com/patcg-individual-drafts/private-aggregation-api/pull/65.html" title="Last updated on Jun 15, 2023, 4:24 PM UTC (4c5f2e4)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/patcg-individual-drafts/private-aggregation-api/65/fbc6b42...4c5f2e4.html" title="Last updated on Jun 15, 2023, 4:24 PM UTC (4c5f2e4)">Diff</a>

### CODE REVIEW

**Comment by alexmturner** [comment]
@yoavweiss I adapted your paragraph into a section in the spec, let me know if you have any thoughts :)
cc @linnan-github

File: spec.bs [hunk 1]
```diff
@@ -91,6 +91,27 @@ batch the reports and send them to the aggregation service for processing. The
...
+Instead of the above API shape, we considered instead aligning with a design
+that is much closer to {{WindowOrWorkerGlobalScope/fetch()}}. However, there are
+a few key differences which made this unfavorable:
+- This API is designed to be used in isolated contexts where
+    {{WindowOrWorkerGlobalScope/fetch()}} is not available.
+- It's an anti-goal to give the developer control over when [=aggregatable
+    reports=] are being sent or knowledge that they were sent (outside of the
+    isolated context). (Except when providing a context ID from <em>outside</em>
+    the isolated context, see [Protecting against leaks via the number of
+    reports](#protecting-against-leaks-via-the-number-of-reports) below.
```
Comment (linnan-github) [inline spec.bs:106]:
```suggestion
    reports](#protecting-against-leaks-via-the-number-of-reports) below.)
```

File: spec.bs [hunk 2]
```diff
@@ -91,6 +91,27 @@ batch the reports and send them to the aggregation service for processing. The
 output of that process is a summary report containing the (approximate) result,
 which is dispatched back to the script's origin.
 
+Considered alternative {#considered-alternative}
+------------------------------------------------
+
+Instead of the above API shape, we considered instead aligning with a design
```
Comment (linnan-github) [inline spec.bs:97]: The API shape below?

Comment (alexmturner) [reply spec.bs]: Reworded to "chosen API shape"

**Review by linnan-github — APPROVED** [review]

**Pull request merged**